### PR TITLE
Method source code

### PR DESCRIPTION
**Pull request by colby-swandale**
## Summary

This PR replaces the "open source in Github" button that's shown on each method with a button that shows the actual source code (which also includes a link to view the source code on Github).

Screen Shot 2020-09-16 at 9 28 57 pm

### Details

This PR adds 2 bits of functionality:

* Import & parse a method's source code from RDoc into ElasticSearch. Parsing is being done by the [Rouge](https://github.com/rouge-ruby/rouge) gem, with some cleanup needed.
* Replace the Github button with a button to toggle viewing a method's source code from directly inside the app. The Github link can be found on the code block header to maintain the existing functionality.

### CODE REVIEW

File: app/javascript/controllers/method_controller.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  toggleShowSource(event) {
+    event.preventDefault()
+    this.showSrcBodyButtonTarget.classList.toggle("bg-blue-600")
+    this.showSrcBodyButtonTarget.classList.toggle("text-white")
+    this.showSrcBodyButtonTarget.classList.toggle("hover:bg-blue-700")
+    this.showSrcBodyButtonTarget.classList.toggle("hover:text-white-300")
+    this.showSrcBodyButtonTarget.classList.toggle("dark:bg-gray-900")
+    this.srcBodyTarget.classList.toggle("hidden")
+  }
+}
```
Comment (AlexWayfer) [inline app/javascript/controllers/method_controller.js:15]: I think we need for a newline at the end of files. I can create a PR with [EditorConfig](https://editorconfig.org/) file, if you wish, for convience.

File: app/views/objects/show/_methods.html.slim [hunk 2]
```diff
@@ -14,8 +14,17 @@ div class="md:px-3 my-3"
...
-        a class="px-1 ml-2 h-6 inline-block rounded bg-gray-200 dark:bg-gray-700 align-middle hover:bg-gray-300 hover:text-gray-800 dark-hover:bg-gray-900 dark-hover:text-gray-400 hover:fill-current" href="#{github_url m}" target="_blank" rel="noopener" title="View source on Github"
-          i class="fab fa-github"
+        button class="px-1 ml-2 h-6 inline-block rounded bg-gray-200 dark:bg-gray-700 align-middle hover:bg-gray-300 dark-hover:bg-gray-900 dark-hover:text-gray-400 hover:fill-current disabled:opacity-50 disabled:cursor-default" title="Show Source Code" data-target="method.showSrcBodyButton" disabled=m.alias? data-action="click->method#toggleShowSource"
+          i class="fas fa-code"
+    div class="hidden my-2" data-target="method.srcBody"
+      div class="w-full px-3 py-2 bg-code-header dark:bg-gray-700 items-center flex justify-between font-mono rounded-t"
+        span class="text-gray-300 text-sm"
+          | #{m.source_file}
+        a class="text-gray-300 text-sm hover:text-gray-400" href="#{github_url m}" target="_blank" rel="noopener" title="View source on Github"
+          | View on Github <i class="fas fa-external-link-alt"></i>
```
Comment (AlexWayfer) [inline app/views/objects/show/_methods.html.slim:24]:
```suggestion
        a class="text-gray-300 text-sm hover:text-gray-400" href="#{github_url m}" target="_blank" rel="noopener" title="View source on GitHub"
          | View on GitHub <i class="fas fa-external-link-alt"></i>
```

File: app/javascript/controllers/method_controller.js (hunk 1 above)
Comment (colby-swandale) [reply app/javascript/controllers/method_controller.js:15]: @AlexWayfer I am happy to add an editor config to file to help maintain consistency 👍 Do you want to make a PR to add that?

Comment (AlexWayfer) [reply app/javascript/controllers/method_controller.js]: > @AlexWayfer I am happy to add an editor config to file to help maintain consistency  Do you want to make a PR to add that?

Yeah, I can. Honestly, it's already unstaged in my local copy of the project, so it should not be difficult.

**Comment by natematykiewicz** [comment]
This is awesome @colby-swandale!

**Pull request merged**